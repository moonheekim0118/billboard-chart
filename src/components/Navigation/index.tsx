import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Navigation = () => {
    return (
        <Container>
            <MenuWrapper>
                <MenuItem to="/">Home</MenuItem>
                <MenuItem to="/findChart">find chart</MenuItem>
            </MenuWrapper>
        </Container>
    );
};

const Container = styled.div`
    position: sticky;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    width: 100%;
    height: 80px;

    background-color: rgba(144, 144, 144, 0.8);
`;

const MenuWrapper = styled.div`
    display: flex;
`;

const MenuItem = styled(Link)`
    color: #fff;
    font-size: 1.4em;
    text-decoration: none;
    font-weight: bold;
    height: 100%;
    padding: 20px 30px 30px 20px;

    transition: background-color 0.5s ease;

    &:hover {
        background-color: #fff;
        text-decoration: underline;
        color: #34495e;
    }
`;

export default Navigation;
