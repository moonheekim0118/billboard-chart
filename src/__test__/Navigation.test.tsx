import 'jsdom-global/register';
import React from 'react';
import Navigation from '../components/Navigation';
import { HashRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('<Navigation/>', () => {
    let container = null;

    it('renders correctly', () => {
        container = mount(
            <HashRouter>
                <Navigation />
            </HashRouter>
        );
    });

    it('matches snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});
