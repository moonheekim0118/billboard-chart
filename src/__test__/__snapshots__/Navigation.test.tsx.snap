// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> matches snapshot 1`] = `"<div class=\\"sc-bdfBwQ hrWjWM\\"><div class=\\"sc-gsTCUz dACEbQ\\"><a class=\\"sc-dlfnbm ldAUts\\" href=\\"#/\\">Home</a><a class=\\"sc-dlfnbm ldAUts\\" href=\\"#/about\\">about</a></div></div>"`;
